restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/Call.java

```diff
@@ -1639,25 +1640,32 @@ private void onSipServletResponse(SipServletResponse message, ActorRef self, Act
 
     private void onHangup(Hangup message, ActorRef self, ActorRef sender) throws Exception {
         if(logger.isDebugEnabled()) {
-            logger.debug("Got Hangup for Call, from: "+from+" to: "+to+" state: "+fsm.state());
+            logger.debug("Got Hangup for Call, from: "+from+" to: "+to+" state: "+fsm.state()+" conferencing: "+conferencing +" conference: "+conference);
         }
-        if (is(updatingMediaSession) || is(ringing) || is(queued) || is(dialing) || is(inProgress)) {
-            if (!receivedBye) {
-                // Send BYE to client if RestComm took initiative to hangup the call
-                sendBye(message);
+
+        // Stop recording if necessary
+        if (recording) {
```

Comment:
Is there a reason to first stop recording and then remove participant or disconnect the call